audacity/audacity
# Time stretching 5 of 6 repair effects and generators for stretched audio

Resolves: #4850

This PR repairs destructive effects for stretched clips.




Recommended:

## QA
All effects are supposed to work. Trying all of them may be too much, but a variety of types of effects should be tried (time-related effect, Nyquist, EQs and filters, noise removal, pitch shift, spectral tools, etc. Of course try them on stretched data, but also on stereo data, etc.

Note that applying an effect on a partial clip selection will create a new clip if the clip was stretched. That's wanted. The old behavior is only preserved on unstretched data.

- [ ] AutoDuck
- [ ] Change Speed
- [ ] Click Removal
- [ ] EQ
- [ ] Generate silence
- [ ] Loudness
- [ ] Noise Reduction
- [ ] Normalize
- [ ] Paulstretch
- [ ] Repair (Since this requires up to 128 samples only, the result will probably only be silence, and that's ok: it's a known limitation of the algorithm that rendering very tiny portions of audio only output silence)
- [ ] Repeat
- [ ] Reverse
- [ ] Change Tempo and Sliding Scale, high and low quality
- [ ] Stereo To Mono (previously reported by @dozzer as https://github.com/audacity/audacity/issues/5118)
- [ ] Truncate Silence
- [ ] Compressor
- [ ] Some Nyquist effects and generators
- [ ] At least one "Per Track Effect" example (any of the realtime capable, built-in or plug-in effects, but used destructively)
